poofarmer/Scripts/animal.gd

```diff
@@ -34,19 +30,31 @@ func _ready():
 
 # Called every frame. 'delta' is the elapsed time since the previous frame.
 func _process(delta):
-	var velocity = Vector2()
-	velocity = (destination - position).normalized() * speed * delta
+	var vector_from_angle = Vector2.RIGHT.rotated(angle).normalized()
+	velocity = vector_from_angle * speed * delta
 	velocity = move_and_slide(velocity)
-	if (position.distance_to(destination) < 100):
-		_on_ChangeDestinationTimer_timeout()
+	# Keep unicorns away from the silo and all animals from getting hung up on it
+	if ((position.distance_to(silo.position) < 1000 && _type == AnimalType.values.Unicorn)) || position.distance_to(silo.position) < 115:
```

Comment:
Not really something we should handle here, but as I'm seeing us make animals behave differently makes me think that we should go the route of of the Enemies and just have sub Nodes for each animal, that way we can not worry about type picking for sprites and distances, but to have each sub animal just implement how _it_ should behave.

I think we can move forward with this though

Reply:
You can just resolve this conversation when you're ready.